graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/RemoteReindexingMigrationAdapterOS2.java

```diff
@@ -126,7 +133,7 @@ void waitForClusterRestart(final Set<String> activeNodes) {
                 .retryIfException()
                 .build();
 
-        final Callable<Boolean> callable = () -> activeNodes.containsAll(getActiveNodes());
+        final Callable<Boolean> callable = () -> getActiveNodes().containsAll(expectedNodes);
```

Comment:
@janheise the original condition evaluated to true even if there were no active nodes available, leading to premature wait termination and errors later in the handling